tests/integration/test_model.py

```diff
@@ -44,153 +36,167 @@ def check_trained_model(c, model_name=None):
+            ) AS (
+                SELECT x, y, x*y > 0 AS target
+                FROM {timeseries}
+            )
+        """
+        )
+        check_trained_model(c, df_name=timeseries)
 
-# TODO - many ML tests fail on clusters without sklearn - can we avoid this?
-@xfail_if_external_scheduler
-def test_training_and_prediction(c, training_df):
     c.sql(
-        """
-        CREATE MODEL my_model WITH (
-            model_class = 'sklearn.ensemble.GradientBoostingClassifier',
-            wrap_predict = True,
+        f"""
+        CREATE OR REPLACE MODEL my_model WITH (
+            model_class = 'LinearRegression',
             target_column = 'target'
         ) AS (
-            SELECT x, y, x*y > 0 AS target
-            FROM timeseries
-            LIMIT 100
+            SELECT x, y, x*y AS target
+            FROM {timeseries}
         )
     """
     )
+    check_trained_model(c, df_name=timeseries)
 
-    check_trained_model(c)
 
+# TODO - many ML tests fail on clusters without sklearn - can we avoid this?
+@xfail_if_external_scheduler
+@pytest.mark.xfail(
+    sys.platform == "win32",
+    reason="'xgboost.core.XGBoostError: Failed to poll' on Windows only",
+)
+@pytest.mark.parametrize(
+    "gpu_client", [False, pytest.param(True, marks=pytest.mark.gpu)], indirect=True
+)
+def test_xgboost_training_prediction(c, gpu_client):
+    gpu = "CUDA" in str(gpu_client.cluster)
+    timeseries = "gpu_timeseries" if gpu else "timeseries"
 
-@pytest.mark.gpu
-def test_cuml_training_and_prediction(c, gpu_training_df):
-    model_query = """
+    # TODO: XGBClassifiers error on GPU
+    if not gpu:
+        c.sql(
+            """
         CREATE OR REPLACE MODEL my_model WITH (
-            model_class = 'cuml.linear_model.LogisticRegression',
-            wrap_predict = True,
-            wrap_fit = False,
+            model_class = 'DaskXGBClassifier',
             target_column = 'target'
         ) AS (
-            SELECT x, y, x*y > 0 AS target
+            SELECT x, y, x*y > 0  AS target
             FROM timeseries
+            LIMIT 100
         )
         """
-    c.sql(model_query)
-    check_trained_model(c)
-
-
-@pytest.mark.gpu
-@xfail_if_external_scheduler
-def test_dask_cuml_training_and_prediction(c, gpu_training_df, gpu_client):
+        )
+        check_trained_model(c)
 
-    model_query = """
+        c.sql(
+            """
         CREATE OR REPLACE MODEL my_model WITH (
-            model_class = 'cuml.dask.linear_model.LinearRegression',
+            model_class = 'XGBClassifier',
             target_column = 'target'
         ) AS (
-            SELECT x, y, x*y AS target
+            SELECT x, y, x*y > 0  AS target
             FROM timeseries
+            LIMIT 100
         )
         """
-    c.sql(model_query)
-    check_trained_model(c)
+        )
+        check_trained_model(c)
 
+    # For GPU tests, set tree_method = 'gpu_hist'
+    tree_method = "gpu_hist" if gpu else "hist"
```

Comment:
As seen here if users are working with gpu df's and haven't specified tree_method `gpu_hist` we end up training the model on the cpu's. 
Do you have any opinions on whether we should warn the users in this case or if it's up to the user to specify these arguments as expected.

Reply:
Personally, I think that type of warning would make more sense coming from the XGBoost side. I could open an issue with them of that's of interest?

If you think we should just raise the warning on the Dask-SQL side instead, though, I'd be happy to add it.

Reply:
No I think it's a good idea to suggest xgboost to warn users when such a situation arises. Okay to not have anything on the dask-sql side